ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@rossberg-chromium Actually, I went back and read your comments again and checked out the original multi-value support again and I see what you're getting at.  Yes, reverting #180 makes sense assuming that we pretend that `expr option` is actually an `expr list` and then encode the list length as an immediate in the binary encoding.  This would be enough for a post-order/bottom-up algo.  To better stake out the ground for multi-values after the MVP, I wonder if we should also go back to having `expr_type` be an `expr list` again (whose lengths were necessarily either 0 or 1) so that we didn't have to wave our hands with the `expr option`...